controllers/api/userRoutes.js

```diff
@@ -5,12 +5,18 @@ router.post('/', async (req, res) => {
   try {
      const Userdata = await User.create(req.body)
 
+     req.session.save(() => {
+        req.session.user_id = Userdata.id;
+        req.session.logged_in = true;
+        res.status(200).json(Userdata);
+     });
+    } catch (err) {
+        res.status(400).json(err);
+        if(User) {
+            res.status(400).json()
+        }
```

Comment:
I don't understand what we are doing here - why would we strip the error if there's a User?